arbnode/inbox_reader.go

```diff
@@ -407,6 +445,18 @@ func (ir *InboxReader) run(ctx context.Context) error {
 			} else {
 				from = arbmath.BigAddByUint(to, 1)
 			}
+			haveMessages := uint64(len(delayedMessages) + len(sequencerBatches))
+			if haveMessages <= (ir.config.TargetMessagesRead / 2) {
+				// This cannot overflow, as it'll never try to subtract more than blocksToFetch
+				blocksToFetch -= (blocksToFetch + 4) / 5
+			} else if haveMessages >= (ir.config.TargetMessagesRead * 3 / 2) {
+				blocksToFetch += (blocksToFetch + 4) / 5
+			}
```

Comment:
I think these are flipped

```suggestion
			if haveMessages >= (ir.config.TargetMessagesRead / 2) {
				// This cannot overflow, as it'll never try to subtract more than blocksToFetch
				blocksToFetch -= (blocksToFetch + 4) / 5
			} else if haveMessages <= (ir.config.TargetMessagesRead * 3 / 2) {
				blocksToFetch += (blocksToFetch + 4) / 5
			}
```

Reply:
Fixed, thanks :) I've also tested with a printf locally that it's fetching large batches as expected (and even that it fetches smaller batches at certain points, presumably when there were a lot of messages)

Reply:
A nova node now syncs in 3 minutes btw